Automattic/jetpack
# WAF: Fix invalid headers

This PR adds a sanitation method to the runtime class. The output that is generated when a request is blocked will now be sanitized before being logged. Further, we will now just return the rule ID in the header response for blocked requests.




Asana [#1201890219548759](https://app.asana.com/0/1201069996155224/1201890219548759)

#### Changes proposed in this Pull Request:

* Adds sanitation_output method to the `class-waf-runtime` class.
* The `class-waf-runtime` block method now uses the sanitize_output method before logging the reason for blocking.
* The block method now only returns the rule ID rather than the reason in the response header.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Update your Jurassic Tube test site's rules.php file with the latest from https://github.com/Automattic/jetpack-waf-rules/blob/trunk/waf-rules/rules.php
* Run `curl -D - -s --url "https://<your-sub-domain>.jurassic.tube/xss" --data "a=<svg/onload%0c=alert%601%60>"`
* You should see a 500 error response with a 403 error log
* Update your site with this PR (ensure rules.php is also re-updated)
* Running the curl no longer generates a 500 error
* Run `curl -D - -s --url "https://<your-sub-domain>.jurassic.tube?jetpack-waf-always-block"`

**Expectation:** The response should now say "403 - rule 1337", the reason should be printed below the response header.
